test-integration/operator-tests/src/test/java/org/infinispan/operator/ClientValidationIT.java

```diff
@@ -0,0 +1,110 @@
+package org.infinispan.operator;
+
+import cz.xtf.builder.builders.SecretBuilder;
+import cz.xtf.client.Http;
+import cz.xtf.core.http.Https;
+import cz.xtf.core.openshift.OpenShift;
+import cz.xtf.core.openshift.OpenShifts;
+import cz.xtf.junit5.annotations.CleanBeforeAll;
+import io.fabric8.kubernetes.api.model.Secret;
+import org.assertj.core.api.Assertions;
+import org.infinispan.Infinispan;
+import org.infinispan.Infinispans;
+import org.infinispan.TestServer;
+import org.infinispan.identities.Credentials;
+import org.infinispan.util.CleanUpValidator;
+import org.infinispan.util.KeystoreGenerator;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+import java.io.IOException;
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Paths;
+import java.util.Map;
+
+/**
+ * ClientValidationIT tests if there is client validation strategy enabled. To validate client certificates,
+ * Data Grid requires a trust store that contains any part of the certificate chain for the signing authority,
+ * typically the root CA certificate. Any client that presents a certificate signed by the CA can connect to Data Grid.
+ */
+@CleanBeforeAll
+class ClientValidationIT {
+   private static final OpenShift openShift = OpenShifts.master();
+   private static final Infinispan infinispan = Infinispans.clientTlsValidation();
+   private static final TestServer testServer = TestServer.get();
+   private static final String testServerHost = testServer.host();
+
+   private static KeystoreGenerator.CertPaths certs;
+   private static String appName;
+   private static String hostName;
+   private static String user;
+   private static String pass;
+
+   @BeforeAll
+   static void deploy() throws IOException {
+      appName = infinispan.getClusterName();
+      hostName = openShift.generateHostname(appName + "-external");
+
+      certs = KeystoreGenerator.generateCerts(hostName, new String[]{appName});
```

Comment:
This are the certificates generated for the Infinispan server, yet they are used by client (test server)

Reply:
Is this wrong? Should be changed? And how?